### DISCUSSION TITLE
Some trouble when running continual-learning-baselines

### DISCUSSION THREAD

**Discussion by August9708** [ContinualAI/avalanche#1544] [Q&A]
Hi,
    I am learning to use Avalanche, and I notice this [repository](https://github.com/ContinualAI/continual-learning-baselines)
https://github.com/ContinualAI/continual-learning-baselines
    I think it may be a great demo and I want to run it. But I meet some trouble about environment config... I report is in the issues of continual-learning-baselines.
https://github.com/ContinualAI/continual-learning-baselines/issues/65#issue-2020212336

> avalanche-lib Version: 0.4.0a0
> 
> When I use: python experiments/split_mnist/synaptic_intelligence.py or python -m unittest tests.SynapticIntelligence.test_smnist
> 
> I get something like: `Traceback (most recent call last): File "/home/avabaseline/continual-learning-baselines/experiments/split_mnist/synaptic_intelligence.py", line 7, in <module> from experiments.utils import set_seed, create_default_args File "/home/avabaseline/continual-learning-baselines/experiments/__init__.py", line 1, in <module> from . import split_mnist File "/home/avabaseline/continual-learning-baselines/experiments/split_mnist/__init__.py", line 1, in <module> from .synaptic_intelligence import synaptic_intelligence_smnist File "/home/avabaseline/continual-learning-baselines/experiments/split_mnist/synaptic_intelligence.py", line 7, in <module> from experiments.utils import set_seed, create_default_args File "/home/avabaseline/continual-learning-baselines/experiments/utils.py", line 9, in <module> from avalanche.benchmarks.utils import AvalancheSubset ImportError: cannot import name 'AvalancheSubset' from 'avalanche.benchmarks.utils' (/home/avalanche/avalanche/benchmarks/utils/__init__.py)`
> 
> avalanche-lib Version: 0.1.0
> 
> When I use: python experiments/split_mnist/synaptic_intelligence.py or python -m unittest tests.SynapticIntelligence.test_smnist
> 
> I get something like: `Traceback (most recent call last): File "experiments/split_mnist/naive.py", line 1, in <module> import avalanche as avl File "/opt/conda/envs/avabase/lib/python3.7/site-packages/avalanche/__init__.py", line 1, in <module> from avalanche import benchmarks File "/opt/conda/envs/avabase/lib/python3.7/site-packages/avalanche/benchmarks/__init__.py", line 13, in <module> from .classic import * File "/opt/conda/envs/avabase/lib/python3.7/site-packages/avalanche/benchmarks/classic/__init__.py", line 1, in <module> from .ccifar10 import * File "/opt/conda/envs/avabase/lib/python3.7/site-packages/avalanche/benchmarks/classic/ccifar10.py", line 19, in <module> from avalanche.benchmarks.datasets import default_dataset_location File "/opt/conda/envs/avabase/lib/python3.7/site-packages/avalanche/benchmarks/datasets/__init__.py", line 11, in <module> from .torchvision_wrapper import * File "/opt/conda/envs/avabase/lib/python3.7/site-packages/avalanche/benchmarks/datasets/torchvision_wrapper.py", line 40, in <module> from torchvision.datasets import Kinetics400 as torchKinetics400 ImportError: cannot import name 'Kinetics400' from 'torchvision.datasets' (/opt/conda/envs/avabase/lib/python3.7/site-packages/torchvision/datasets/__init__.py)`
> 
> Okay, something wrong with torch version, pip install torch==1.9.0 torchvision==0.10.0
> 
> Then new problem is : `Traceback (most recent call last): File "/home/avabaseline/continual-learning-baselines/experiments/split_mnist/naive.py", line 7, in <module> from experiments.utils import set_seed, create_default_args File "/home/avabaseline/continual-learning-baselines/experiments/__init__.py", line 1, in <module> from . import split_mnist File "/home/avabaseline/continual-learning-baselines/experiments/split_mnist/__init__.py", line 3, in <module> from .gss import gss_smnist File "/home/avabaseline/continual-learning-baselines/experiments/split_mnist/gss.py", line 3, in <module> from avalanche.benchmarks import CLExperience ImportError: cannot import name 'CLExperience' from 'avalanche.benchmarks' (/opt/conda/envs/ab/lib/python3.9/site-packages/avalanche/benchmarks/__init__.py)`
> 
> `I also try other version of avalanche-lib. And different errors happened, such as: ImportError: cannot import name 'MIRPlugin' from 'avalanche.training.plugins' (/opt/conda/envs/ab/lib/python3.9/site-packages/avalanche/training/plugins/__init__.py)`
> 
> OMG！ I don't know how to solve it and I'm upset. Maybe I made some silly mistakes. Please forgive me and help me. Give me some details about correct envinronments plz. THX!!!

   How can I run it successfully?

**Comment by WanyiLi** [comment]
I met the same problem also.
